services/director-v2/src/simcore_service_director_v2/models/schemas/clusters.py

```diff
@@ -70,14 +71,16 @@ class WorkersDict(DictModel[AnyUrl, Worker]):
 
 class Scheduler(BaseModel):
     status: str = Field(..., description="The running status of the scheduler")
-    workers: WorkersDict | None = Field(default_factory=dict)
+    workers: WorkersDict | None = Field(
```

Comment:
can you replace this fix with the following?
```python
WorkersDict: TypeAlias = dict[AnyUrl, Worker]


class Scheduler(BaseModel):
    status: str = Field(..., description="The running status of the scheduler")
    workers: WorkersDict | None = Field(default_factory=dict)

    @validator("workers", pre=True, always=True)
    @classmethod
    def ensure_workers_is_empty_dict(cls, v):
        if v is None:
            return {}
        return v
```

I already fixed it in a separate PR, but since you're at it. Thanks

Reply:
also change then accordinglu in test_models_clusters.py:
```python
def test_scheduler_constructor_with_default_has_correct_dict(faker: Faker):
    scheduler = Scheduler(status=faker.text())
    assert scheduler.workers is not None
    assert len(scheduler.workers) == 0


def test_scheduler_constructor_with_no_workers_has_correct_dict(faker: Faker):
    scheduler = Scheduler(status=faker.text(), workers=None)
    assert scheduler.workers is not None
    assert len(scheduler.workers) == 0
```

thanks!

Reply:
thanks for this!